src/client/mod.rs

```diff
@@ -612,7 +616,7 @@ impl<T> VaultClient<T>
     ///
     /// [token]: https://www.vaultproject.io/docs/auth/token.html
     pub fn renew(&mut self) -> Result<()> {
-        let res = try!(self.post("/v1/auth/token/renew-self", None, None));
+        let res = try!(self.post::<_, String>("/v1/auth/token/renew-self", None, None));
```

Comment:
Annoying to have to do these turbofish everywhere... Too bad it can't figure out some type for `None`. I've ran into this many times too, I don't see any way to avoid it through.